### PR TITLE
Feature/follow

### PR DESCRIPTION
**Pull request by VILGITU**
Fixed a minor bug. 

The follow functionality now works but only under the assumption that a user has written a post (and therefore has an author profile associated to their account. This will possibly be fixed with Marcus' change regarding user/authors?

### CODE REVIEW

**Review by marcusaandahl — CHANGES_REQUESTED** [review]
Few changes, mainly about regarding cleaning your code of debug logs and following the project structure & features. Otherwise nice implementation.

File: src/Chirp.Core/Repositories/ICheepRepository.cs [hunk 1]
```diff
@@ -8,5 +8,15 @@ public interface ICheepRepository
     public int GetAuthorCheepCount(string authorName);
     public List<CheepDto> GetCheepsForPage(int pageNumber);
     public List<CheepDto> GetAuthorCheepsForPage(string authorName, int pageNumber);
+
+    public List<CheepDto> GetAuthorCheepsForPageAsOwner(string authorName, int pageNumber);
+    
     public bool DeleteCheep(String cheepId, String author);
+    public List<string> GetFollowsForAuthor(string authorEmail);
```
[the remark this answers is not in the document]
Comment (VILGITU) [reply src/Chirp.Core/Repositories/ICheepRepository.cs:15]: Made issue #195

File: src/Chirp.Infrastructure/Repositories/CheepRepository.cs [hunk 2]
```diff
@@ -102,6 +102,33 @@ public List<CheepDto> GetAuthorCheepsForPage(string authorName, int pageNumber)
         });
     }
 
+    public List<CheepDto> GetAuthorCheepsForPageAsOwner(string authorName, int pageNumber)
+    {
+        List<string> authorFollows = GetFollowsForAuthor(GetAuthorEmailByName(authorName));
```
[the remark this answers is not in the document]
Comment (VILGITU) [reply src/Chirp.Infrastructure/Repositories/CheepRepository.cs:107]: Made issue #196

File: src/Chirp.Infrastructure/Repositories/CheepRepository.cs [hunk 3]
```diff
@@ -129,4 +156,96 @@ public bool DeleteCheep(String cheepId, String author)
...
+            
+            userAuthor.Follows.Add(followAuthor);
+
+            followAuthor.FollowedBy.Add(userAuthor);
+
+            _chirpDbContext.SaveChanges();
+        }
+        catch (Exception e)
+        {
+
```
[the remark this answers is not in the document]
Comment (VILGITU) [reply src/Chirp.Infrastructure/Repositories/CheepRepository.cs:202]: Removed try-catch and returns void instead

**Review by marcusaandahl — APPROVED** [review]

**Pull request merged**